framework/include/userobject/UserObject.h

```diff
@@ -99,6 +99,17 @@ class UserObject : public MooseObject,
     mooseError(name(), " does not satisfy the Spatial UserObject interface!");
   }
 
+  /**
+   * Optional interface function for providing the points at which a UserObject attains
+   * spatial values. If a UserObject overrides this function, then other objects that
+   * take both the UserObject and points can instead directly use the points specified
+   * on the UserObject.
+   */
+  virtual const std::vector<Point> spatialPoints() const
+  {
+    mooseError(name(), " does not satisfy the points Spatial UserObject interface!");
```

Comment:
```suggestion
    mooseError("Spatial UserObject interface is not satisfied; spatialPoints() must be overridden");
```

You don't need to use `name()` in errors anymore; the object type and name are printed as a prefix for the error